api/api-internal-environments-v1/src/main/java/com/thoughtworks/go/apiv1/internalenvironments/InternalEnvironmentsControllerV1.java

```diff
@@ -108,17 +109,15 @@ public String indexMergedEnvironments(Request request, Response response) throws
     String updateAgentAssociation(Request request, Response response) {
         String envName = request.params("env_name");
         EnvironmentConfig envConfig = environmentConfigService.getEnvironmentConfig(envName);
-        List<String> uuids = getAgentUuids(request);
-        agentService.updateAgentsAssociationOfEnvironment(envConfig, uuids);
-        return renderMessage(response, 200, "Environment '" + envName + "' updated successfully!");
-    }
-
-    private List<String> getAgentUuids(Request request) {
         JsonReader jsonReader = GsonTransformer.getInstance().jsonReaderFrom(request.body());
-        Optional<List<String>> uuids = jsonReader.readStringArrayIfPresent("uuids");
-        if (uuids.isPresent()) {
-            return uuids.get();
+        JsonReader agents = jsonReader.readJsonObject("agents");
+
+        List<String> uuidsToAssociate = agents.readStringArrayIfPresent("add").orElse(Collections.emptyList());
+        List<String> uuidsToRemove = agents.readStringArrayIfPresent("remove").orElse(Collections.emptyList());
+
+        if (!uuidsToAssociate.isEmpty() || !uuidsToRemove.isEmpty()) {
+            agentService.updateAgentsAssociationOfEnvironment(envConfig, uuidsToAssociate, uuidsToRemove);
```

Comment:
@vrushaliwaykole @dhanasp -- There is a null check inside `updateAgentsAssociationOfEnvironment` as there is only one usage of this method and we're already making sure that the environment exists as part of `environmentConfigService.getEnvironmentConfig` should we remove the null check verification from `updateAgentsAssociationOfEnvironment` ?

Reply:
@GaneshSPatil agreed that the method `updateAgentsAssociationOfEnvironment` is currently only used in the Internal controller. But is that enough to remove the check.
The service can be called from anywhere (in future) and shouldn't it be the responsibility of the method itself to handle all the cases? WDYT?
cc:\ @viraj2712

Reply:
There should be no reason a `null` is sent for environment config. I couldn't think of any scenario where an environment is null, hence the thought of removing that check.